tmt/steps/provision/minute.py

```diff
@@ -0,0 +1,250 @@
+import base64
+import datetime
+import getpass
+import json
+import os
+import re
+import time
+
+import click
+import requests
+from urllib3.exceptions import InsecureRequestWarning
+
+import tmt
+from tmt.utils import RunError, ProvisionError
+
+
+DEFAULT_USER = 'root'
+SSH_KEY = '/usr/share/qa-tools/1minutetip/1minutetip'
+NUMBER_OF_RETRIES = 3
+DEFAULT_CONNECT_TIMEOUT = 60
+
+
+def run_openstack(url, cmd, cached_list=False):
+    """
+    Runs an openstack command.
+    
+    Returns (exit_code, stdout) tuple. Both are None if the request failed.
+    """
+    url += '/openstack.php'
+    data = {
+        'CMD': base64.b64encode(cmd.encode('ascii')),
+        'base64': 1,
+        }
+    if cached_list:
+        data['use_cached_list'] = 1
+    # Disable warning about insecure connection. Using insecure connection
+    # is unfortunately necessary here for the plugin to work.
+    requests.packages.urllib3.disable_warnings(category=InsecureRequestWarning)
+    r = requests.post(url, verify=False, data=data)
+    if r.ok:
+        # The output is in the form of: <stdout>\n<exit>\n.
+        split = r.text.rsplit('\n', 2)
+        return int(split[1]), split[0]
+    return None, None
+
+
+class Provision1minutetip(tmt.steps.provision.ProvisionPlugin):
+    """
+    Use 1minutetip's openstack backend script to provision a guest
+
+    Minimal configuration using the latest Fedora image:
+        
+        provision:
+            how: 1minutetip
+
+    Full configuration example:
+        
+        provision:
+            how: 1minutetip
+            image: 1MT-Fedora-32
+            flavor: m1.large
+
+    Possible images and flavors can be found using 1minutetip
+    (see 1minutetip list and 1minutetip list-flavors).
+
+    """
+    # Guest instance
+    _guest = None
+
+    # Supported methods
+    _methods = [
+        tmt.steps.Method(name='1minutetip', doc=__doc__, order=50),
```

Comment:
What about using `minute` for the method name? It would be a bit easier to write on the command line and consistent with the module name.

Reply:
Changed the method name and also names of the classes to make the naming consistent